m1 wheels generation

#### Describe your feature request

Just realize one of my colleague can't install polars on his mac m1. Short of compiling from source on his machine,

what would be the best way of cross-compiling the project for him ?

Is there a plan to support building the m1 wheels for pip ?

I've looked into cibuildwheel and found that it requires a setup.py entrypoint.
Though this works with setup-tools-rust I have yet to find a way to do it with maturin. 

An alternative would be to use the 
```
maturin build --universal2 --release  
``` 
- polars-0.9.7-cp36-abi3-macosx_10_9_x86_64.macosx_11_0_arm64.macosx_10_9_universal2.whl
flag exposed by maturin build/publish that creates an arm64 compatible wheel for apple silicon. I did this on my x86 mbpro and generated a wheel my colleague could install.

I tried to implement that with a github action 

```
name: Create universal2 python release

on: [workflow_dispatch]

jobs:
  build:
    name: Create Release
    runs-on: ${{ matrix.os }}
    strategy:
      matrix:
        os: ["macos-latest"]
        python-version: ["3.6"]
    steps:
        - uses: actions/checkout@v2
        - name: Install latest Rust nightly
          uses: actions-rs/toolchain@v1
          with:
            toolchain: nightly-2021-09-14
            override: true
            components: rustfmt, clippy
        - name: Setup universal2 targets for Rust
          run: |
            rustup target add aarch64-apple-darwin
        - name: Set up Python
          uses: actions/setup-python@v2
          with:
            python-version: ${{ matrix.python-version }}
        - name: Install dependencies
          run: |
            python -m pip install --upgrade pip
            pip install maturin==0.11.0
        - name: Publish wheel
          shell: bash
          env:
            MATURIN_PASSWORD: ${{ secrets.PYPI_PASS }}
            ARCHFLAGS: -arch x86_64 -arch arm64
          run: |
            rm py-polars/README.md
            cp README.md py-polars/README.md
            cd py-polars
            rustup override set nightly-2021-09-14
            maturin publish \
            --universal2 \
            --skip-existing \
            -o wheels \
            -i python \
            -r https://test.pypi.org/legacy/ \
            --username theophaniel \
```

But the workflows fails during the build.

```
   Compiling regex v1.5.4
   Compiling paste v0.1.18
   Compiling signal-hook v0.1.17
   Compiling lexical-core v0.8.0
   Compiling crossbeam-deque v0.8.1
   Compiling rand_chacha v0.2.2
   Compiling pyo3-macros-backend v0.14.5
   Compiling zstd-sys v1.6.1+zstd.1.5.0
   Compiling lz4-sys v1.9.2
   Compiling libmimalloc-sys v0.1.22
   Compiling chrono v0.4.19
   Compiling ndarray v0.15.3
   Compiling thiserror-impl v1.0.29
   Compiling async-stream-impl v0.3.2
   Compiling multiversion-macros v0.6.1
   Compiling strum_macros v0.20.1
   Compiling ghost v0.1.2
   Compiling inventory-impl v0.1.10
   Compiling indoc-impl v0.3.6
   Compiling ctor v0.1.21
   Compiling crossterm v0.19.0
   Compiling lexical v6.0.0
   Compiling rand v0.7.3
   Compiling pyo3-macros v0.14.5
The following warnings were emitted during compilation:

warning: In file included from c_src/mimalloc/src/static.c:17:
warning: In file included from c_src/mimalloc/include/mimalloc-internal.h:11:
warning: In file included from c_src/mimalloc/include/mimalloc-types.h:12:
warning: In file included from /Applications/Xcode_12.4.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/12.0.0/include/stdint.h:52:
warning: In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/stdint.h:52:
warning: In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/sys/_types.h:32:
warning: /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/sys/cdefs.h:807:2: error: Unsupported architecture
warning: #error Unsupported architecture
warning:  ^
warning: In file included from c_src/mimalloc/src/static.c:17:
warning: In file included from c_src/mimalloc/include/mimalloc-internal.h:11:
warning: In file included from c_src/mimalloc/include/mimalloc-types.h:12:
warning: In file included from /Applications/Xcode_12.4.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/12.0.0/include/stdint.h:52:
warning: In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/stdint.h:52:
warning: In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/sys/_types.h:33:
warning: /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/machine/_types.h:34:2: error: architecture not supported
warning: #error architecture not supported
warning:  ^
warning: In file included from c_src/mimalloc/src/static.c:17:
warning: In file included from c_src/mimalloc/include/mimalloc-internal.h:11:
warning: In file included from c_src/mimalloc/include/mimalloc-types.h:12:
warning: In file included from /Applications/Xcode_12.4.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/12.0.0/include/stdint.h:52:
warning: In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/stdint.h:52:
warning: /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/sys/_types.h:55:9: error: unknown type name '__int64_t'
warning: typedef __int64_t       __darwin_blkcnt_t;      /* total blocks */
...
```
So not exacly sure what differs here from what i'm doing on my device so any input from someone proficient in github-action / cross compilation would be welcomed.

full logs here:
https://github.com/tiphaineruy/polars/runs/3623712608?check_suite_focus=true

Migrating to https://github.com/PyO3/setuptools-rust + https://cibuildwheel.readthedocs.io/en/stable/ could be an "easy" solution.

--- a/.github/workflows/create-py-universal2-release.yaml
+++ b/.github/workflows/create-py-universal2-release.yaml
@@ -1,0 +1,41 @@
+name: Create universal2 python release
+
+on: [workflow_dispatch]
+
+jobs:
+  build:
+    name: Create Release
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: ["macos-latest"]
+        python-version: ["3.6"]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install latest Rust nightly
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: nightly-2021-09-14
+          override: true
+          components: rustfmt, clippy
+      - name: Setup universal2 targets for Rust
+        run: |
+          rustup target add aarch64-apple-darwin
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Prepare maturin publish
+        shell: bash
+        run: |
+          rm py-polars/README.md
+          cp README.md py-polars/README.md
+          cd py-polars
+          rustup override set nightly-2021-09-14
+      - uses: maturin publish
+        env:
+          MATURIN_PASSWORD: ${{ secrets.PYPI_PASS }}
+        with:
+          maturin-version: 0.11.3
+          command: publish
+          args: -m py-polars/Cargo.toml --no-sdist --universal2 -o wheels -i python -u ritchie46

